services/user/Branding/ui/src/App.tsx

```diff
@@ -0,0 +1,214 @@
+import { useEffect, useRef, useState } from "react";
+
+import { Button } from "@shadcn/button";
+import { Label } from "@shadcn/label";
+import { Input } from "@shadcn/input";
+
+import { siblingUrl, Supervisor } from "@psibase/common-lib";
+import { Nav } from "@components/nav";
+
+const supervisor = new Supervisor();
+
+export const App = () => {
+    const [changesMade, setChangesMade] = useState<boolean>(false);
+    const [networkName, setNetworkName] = useState<string>("");
+    const [logoBytes, setLogoBytes] = useState<Uint8Array>(new Uint8Array());
+    const fileInput = useRef<HTMLInputElement>(null);
+    const [previewImgUrl, setPreviewImgUrl] = useState<string>("");
+    const [uploadStatus, setUploadStatus] = useState<string>("");
+
+    const init = async () => {
+        await supervisor.onLoaded();
+        supervisor.preLoadPlugins([{ service: "branding" }]);
+        setTimeout(getNetworkName, 1000);
+    };
+
+    useEffect(() => {
+        init();
+    }, []);
+
+    const getNetworkName = async () => {
+        const queriedNetworkName = (await supervisor.functionCall({
+            service: "branding",
+            intf: "queries",
+            method: "getNetworkName",
+            params: [],
+        })) as string;
+        setNetworkName(queriedNetworkName);
+    };
+    const updateAssets: React.MouseEventHandler<HTMLButtonElement> = async (
+        e,
+    ) => {
+        e.preventDefault();
+        try {
+            let res = await supervisor.functionCall({
+                service: "accounts",
+                intf: "accounts",
+                method: "loginTemp",
+                params: ["branding"],
+            });
+
+            if (networkName) {
+                res = await supervisor.functionCall({
+                    service: "branding",
+                    intf: "api",
+                    method: "setNetworkName",
+                    params: [networkName],
+                });
+            }
+
+            if (logoBytes.length) {
+                res = await supervisor.functionCall({
```

Comment:
doesn't matter in this case. This allows for only one item to be updated without complication.
If this were a higher-profile, more-frequently used app, we'd have a single action that takes both netname and logoimage with some means of indicating no update, and it would put both in a single tx.